core/src/Streamly/Internal/Data/Stream/Type.hs

```diff
@@ -173,7 +199,396 @@ instance Monad m => Applicative (Stream m) where
+-- Construction
+------------------------------------------------------------------------------
+
+-- | Build a stream by unfolding a /monadic/ step function starting from a
+-- seed.  The step function returns the next element in the stream and the next
+-- seed value. When it is done it returns 'Nothing' and the stream ends. For
+-- example,
+--
+-- @
+-- let f b =
+--         if b > 3
+--         then return Nothing
+--         else print b >> return (Just (b, b + 1))
+-- in drain $ unfoldrM f 0
+-- @
+-- @
+--  0
+--  1
+--  2
+--  3
+-- @
+--
+-- /Pre-release/
+--
+{-# INLINE toStreamD #-}
+toStreamD :: Applicative m => Stream m a -> D.Stream m a
+toStreamD (Stream m) = D.fromStreamK m
+
+{-# INLINE fromStreamD #-}
+fromStreamD :: Monad m => D.Stream m a -> Stream m a
+fromStreamD m = Stream $ D.toStreamK m
+
+{-# INLINE_EARLY fromList #-}
+fromList :: Monad m => [a] -> Stream m a
+fromList = fromStreamS . S.fromList
+
+-- | Convert a stream into a list in the underlying monad.
+--
+-- @since 0.1.0
+{-# INLINE toList #-}
+toList :: Monad m => Stream m a -> m [a]
+toList m = S.toList $ toStreamS m
+
+-------------------------------------------------------------------------------
+-- Construction
+-------------------------------------------------------------------------------
+
+infixr 5 `cons`
+
+-- | Construct a stream by adding a pure value at the head of an existing
+-- stream. Same as the following but more efficient:
+--
+-- For example:
+--
+-- >> s = 1 `Stream.cons` 2 `Stream.cons` 3 `Stream.cons` Stream.nil
+-- >> Stream.fold Fold.toList s
+-- >[1,2,3]
+--
+-- >>> cons x xs = return x `Stream.consM` xs
+--
+-- /Pre-release/
+--
+{-# INLINE_NORMAL cons #-}
+cons ::  a -> Stream m a -> Stream m a
+cons a (Stream m) = Stream $ K.cons a m
+
+-- | Constructs a stream by adding a monadic action at the head of an
+-- existing stream. For example:
+--
+-- >> s = putChar 'h' `Stream.consM` putChar 'i' `Stream.consM` Stream.nil
+-- >> Stream.fold Fold.toList s
+-- >hi
+--
+-- >> consM x xs = Stream.fromEffect x `Stream.append` xs
+--
+-- /Pre-release/
+--
+{-# INLINE consM #-}
+{-# SPECIALIZE consM :: IO a -> Stream IO a -> Stream IO a #-}
+consM :: Monad m => m a -> Stream m a -> Stream m a
+consM m (Stream ms) = Stream $ K.consM m ms
+
+infixr 5 .:
+
+-- | Operator equivalent of 'cons'.
+--
+-- @
+-- > toList $ 1 .: 2 .: 3 .: nil
+-- [1,2,3]
+-- @
+--
+-- @since 0.1.1
+{-# INLINE (.:) #-}
+(.:) ::  a -> Stream m a -> Stream m a
+(.:) = cons
+
+{-# INLINE_NORMAL nil #-}
+nil ::  Stream m a
+nil = Stream K.nil
+
+{-# INLINE_NORMAL nilM #-}
+nilM :: (Monad m) => m b -> Stream m a
+nilM = Stream . K.nilM
+
+
+{-# INLINE_NORMAL fromPure #-}
+fromPure :: a -> Stream m a
+fromPure = Stream . K.fromPure
+
+{-# INLINE_NORMAL fromEffect #-}
+fromEffect :: Monad m => m a -> Stream m a
+fromEffect = Stream . K.fromEffect
+
+{-# INLINE repeat #-}
+repeat :: a -> Stream m a
+repeat = Stream . K.repeat
```

Comment:
This was removed in master branch in the earlier commit. Should be remain removed.